js/customize-snapshots-front.js

```diff
@@ -0,0 +1,72 @@
+/* global jQuery, wp */
+/* exported CustomizeSnapshotsFront */
+var CustomizeSnapshotsFront = (function( $ ) {
+	'use strict';
+
+	var component = {
+		data: {
+			confirmationMsg: '',
+			action: '',
+			snapshotsFrontendPublishNonce: '',
+			errorMsg: ''
+		}
+	};
+	/**
+	 * Init.
+	 *
+	 * @param {object} args Args.
+	 * @returns {void}
+	 */
+	component.init = function init( args ) {
+		_.extend( component.data, args );
+
+		$( document ).ready( function() {
+			component.setupPublishButton();
+		} );
+	};
+
+	/**
+	 * Set up snapshot frontend publish button.
+	 *
+	 * @returns {void}
+	 */
+	component.setupPublishButton = function setupPublishButton() {
+		var publishBtn = $( '#wp-admin-bar-publish-customize-snapshot a' );
+
+		if ( ! publishBtn.length ) {
+			return;
+		}
+
+		publishBtn.click( function( e ) {
+			var request,
+				data = {
+					nonce: component.data.snapshotsFrontendPublishNonce,
+					uuid: component.data.uuid
+				};
+			e.preventDefault();
+
+			if ( ! window.confirm( component.data.confirmationMsg ) ) { // eslint-disable-line no-alert
+				return false;
+			}
+			if ( ! wp.customize.settings.theme.active ) {
```

Comment:
This `wp.customize.settings` value is actually set in `customize-preview` (and `customize-controls`) not `customize-base`. 

See: https://github.com/WordPress/wordpress-develop/blob/4.8.0/src/wp-includes/js/customize-preview.js#L657-L658

Note that it is set at `jQuery.ready`.

But since you'd be clicking the button after `jQuery.ready` anyway, I don't see why you it would be undefined. Is `customize-preview` itself not getting enqueued? It should be if you've loaded a changeset on the frontend.